packages/manager/modules/billing-components/src/components/services-actions/services-actions.controller.js

```diff
@@ -112,4 +117,39 @@ export default class ServicesActionsCtrl {
       this.handleGoToResiliation();
     }
   }
+
+  /**
+   * Adds a 'click' event listener in capture mode on the actionMenu element.
+   *
+   * The <oui-action-menu> component does not support an 'on click' handler,
+   * making it impossible to catch clicks directly through its attributes.
+   * To work around this, the listener is added in capture mode.
+   * This way, it catches the click before <oui-action-menu> can call stopPropagation or preventDefault,
+   * ensuring that the getPendingEngagement() function is triggered on click.
+   */
+  initActionMenuClick() {
+    const actionMenu = this.$element[0];
+    actionMenu.addEventListener(
+      'click',
+      () => {
+        this.getPendingEngagement();
+      },
+      true,
+    );
+  }
+
+  getPendingEngagement() {
+    if (!this.loadedPendingEngameent) {
+      this.BillingService.getPendingEngagement(this.service.id)
+        .then(() => {
+          this.service.hasPendingEngagement = true;
+        })
+        .catch(() => {
+          this.service.hasPendingEngagement = false;
+        })
+        .finally(() => {
+          this.loadedPendingEngameent = true;
+        });
+    }
```

Comment:
```suggestion
  getPendingEngagement() {
    if (this.service.hasPendingEngagement !== undefined) {
      this.BillingService.getPendingEngagement(this.service.id)
        .then(() => {
          this.service.hasPendingEngagement = true;
        })
        .catch(() => {
          this.service.hasPendingEngagement = false;
        });
    }
```
If possible use directly `this.service.hasPendingEngagement`to check if it's loaded ?

Reply:
No, because the default value of hasPendingEngagement is false !